docs/roadmap.md

```diff
@@ -0,0 +1,184 @@
+# Roadmap
+
+## Aim of the project
+
+* Enable application developers to connect their workloads to backing services **with little or no code changes**.
+
+* Enable backing service authors to expose binding information by adding metadata to the representative Kubernetes objects **with little or no code changes**.
+
+* Provide a managed binding experience.
+
+The Service Binding Operator is being designed to be adopted by application developers and backing service providers with a **low barrier to entry**.
+
+
+# Key feature areas
+
+### Collection of binding information
+* Annotation "Decorators" in backing service resources.
+* Descriptor "Decorators" in backing service resources.
+* Duck type in backing service resources.
+* Service Composition.
+* Detection of binding resources.
+
+### Projection of binding information
+* Custom binding variable generation.
+* Projection of binding information in both PodSpec and non-PodSpec based workloads.
+
+### Security
+* Avoiding escalation of privileges.
+* Narrow down the service account privileges.
+
+### Release & Packaging
+* Packaging of the Service Binding Controller/CRD.
+
+----
+
+
+# Collection of binding information
+
+## Annotation "Decorators" in backing service resources
+
+Supporting annotations as a way to decorate kubernetes resources enables the following:
+
+* Application developers can use any kubernetes resource a potential backing service
+* Backing service resources automatically contribute binding information extracted according to specific annotations in their respective CRDs; such annotations could be provided by default by the backing service authors, or by users to any CRD present in the system.
+
+The project aims to support an annotation format such that there's a way to indicate whether a specific unit of binding information is to be bound as a volume mount or an environment variable.
+
+## Improvements
+
+Validate advanced volume mounting support for projecting sensitive information as "files".
+
+
+## OLM Descriptor "Decorators" in backing service resources
+
+Use of OLM Descriptors as a way to indicate what is interesting for binding enables operator-backed services to add metadata outside of their CRDs to indicate what needs to be extracted for binding.
+
+## Improvements
+
+Validate advanced volume mounting support for projecting sensitive information as "files".
+
+## Duck type in backing service resources
+
+
+Identify the information that a duck typed resource needs to contain and 
+support binding secret generation based on the same.
+
+( needs more information )
```

Comment:
Wish to keep this area undefined till we have an enhancement proposal of what the duck typing should look like.

Reply:
Actually, ignore, added this bit.